fluss-flink/fluss-flink-common/src/main/java/org/apache/fluss/flink/source/FlinkTableSource.java

```diff
@@ -572,7 +576,11 @@ && hasPrimaryKey()
                     }
                 }
             }
-            return Result.of(acceptedFilters, remainingFilters);
+
+            // We cannot determine whether this source will ultimately be used as a scan source or
+            // a lookup source. Since fluss lookup sources cannot accept filters yet, to be safe, we
+            // return all filters to the Flink planner.
+            return Result.of(acceptedFilters, filters);
```

Comment:
IIUC, it'll cause the plan still have the accepted filters?

Reply:
I have created a jira in Flink to follow this issue. https://issues.apache.org/jira/browse/FLINK-38635